src/UIState/SeePIDConstants.cpp

```diff
@@ -4,12 +4,72 @@
 
 #include "SeePIDConstants.h"
 
-void SeePIDConstants::handleKey(char key) {
-  switch (key) {
-    case 'D':  // Don't save (cancel)
-      returnToMainMenu();
+#include "Devices/LiquidCrystal_TC.h"
+#include "Devices/PHProbe.h"
+#include "Devices/PID_TC.h"
+#include "TC_util.h"
+
+void SeePIDConstants::loop() {
+  // Precheck
+  if ((msEnd - millis()) <= (1000 * secondsLeft)) {
+    char buffer[100];
+    sprintf(buffer, "Rotate Display Time=%i", msEnd - millis());
+    COUT(buffer);
+    secondsLeft--;
+  }
+
+  switch (secondsLeft) {
+    case 4:
+      loadKp(0);
+      loadKi(1);
+      break;
+    case 3:
+      loadKi(0);
+      loadKd(1);
+      break;
+    case 2:
+      loadKd(0);
+      loadSlope(1);
       break;
-    default:
+    case 1:
+      loadSlope(0);
+      loadKp(1);
       break;
-  };
+    case 0:
+      loadKp(0);
+      loadKi(1);
+      break;
+  }
+
+  if (msEnd < millis()) {
+    COUT("done");
+    returnToMainMenu();
+  }
+}
+
+void SeePIDConstants::loadKp(int line) {
+  PID *pPID = PID_TC::instance()->getPID();
+  char buffer[17];
+  sprintf(buffer, "kp:%.1f", pPID->GetKp());
+  LiquidCrystal_TC::instance()->writeLine(buffer, line);
```

Comment:
I suspect that the LCD does not clear the rest of the line. So far we've been making sure that everything we send to the LCD is exactly 16 characters long and this seems like it is putting the effort in the wrong place. I think that we should modify LiquidCrystal_TC::writeLine() so that it either clears the line first or adds spaces to the end of the string to be printed (copying into an adequate buffer first.

Reply:
See #80.